pkg/aws/eni/node.go

```diff
@@ -538,3 +539,49 @@ func (n *Node) GetMaximumAllocatableIPv4() int {
 	// Return the maximum amount of IP addresses allocatable on the instance
 	return (limits.Adapters - firstInterfaceIndex) * limits.IPv4
 }
+
+// GetMinimumAllocatableIPv4 returns the minimum amount of IPv4 addresses that
+// must be allocated to the instance.
+func (n *Node) GetMinimumAllocatableIPv4() int {
+	n.mutex.RLock()
+	defer n.mutex.RUnlock()
+
+	minimum := defaults.IPAMPreAllocation
+
+	if n.k8sObj == nil || n.k8sObj.Spec.ENI.FirstInterfaceIndex == nil {
+		n.loggerLocked().WithFields(logrus.Fields{
+			"adaptors-limit": "unknown",
+			"pre-allocate":   minimum,
+		}).Warning("Could not determine first-interface-index, falling back to default pre-allocate value")
+		return minimum
+	}
+
+	index := *n.k8sObj.Spec.ENI.FirstInterfaceIndex
```

Comment:
Is this safe? Can't `FirstInterfaceIndex` be nil?

Reply:
Yes, there is a check for it already a few lines above.